Sources/DeepLearning/Operators/Basic.swift

```diff
@@ -0,0 +1,710 @@
+
+public extension TensorFlowScalar {
+    /// Convert to a tensor with the specified rank, with all dimensions equal to 1.
+    @inlinable
+    func makeTensor(rank: Int) -> Tensor<Self> {
+        return Tensor(repeating: self, shape: TensorShape(rank))
+    }
+}
+
+public extension Tensor {
+    /// Unpacks the given dimension of a rank-`R` tensor into multiple rank-`(R-1)` tensors. Unpacks 
+    /// `N` tensors from this tensor by chipping it along the `axis` dimension, where `N` is
+    /// inferred from this tensor's shape. For example, given a tensor with shape `[A, B, C, D]`:
+    /// 
+    ///   - If `axis == 0` then the `i`-th tensor in the returned array is the slice 
+    ///     `self[i, :, :, :]` and each tensor in that array will have shape `[B, C, D]`. 
+    ///     (Note that the dimension unpacked along is gone, unlike
+    ///     `Tensor.split(numSplits:alongAxis)`, or `Tensor.split(sizes:alongAxis)`).
+    ///   - If `axis == 1` then the `i`-th tensor in the returned array is the slice 
+    ///     `value[:, i, :, :]` and each tensor in that array will have shape `[A, C, D]`.
+    ///   - Etc.
+    ///
+    /// This is the opposite of `Tensor.init(stacking:alongAxis:)`.
+    ///
+    /// - Parameters:
+    ///   - axis: Dimension along which to unstack. Negative values wrap around.
+    /// 
+    /// - Precondition: `axis` must be in the range `[-rank, rank)`, where `rank` is the rank of the
+    ///   provided tensors.
+    /// 
+    /// - Returns: Array containing the unstacked tensors.
+    @inlinable
+    @differentiable(vjp: _vjpUnstack(alongAxis:) where Scalar: TensorFlowFloatingPoint)
+    func unstack(alongAxis axis: Int = 0) -> [Tensor] {
+        return Raw.unpack(value: self, num: Int64(shape[axis]), axis: Int64(axis))
+    }
+
+    /// Splits a tensor into multiple tensors. The tensor is split along dimension `axis` into
+    /// `numSplits` smaller tensors. This requires that `numSplits` evenly divides `shape[axis]`.
+    ///
+    /// For example:
+    /// ```
+    /// // 'value' is a tensor with shape [5, 30]
+    /// // Split 'value' into 3 tensors along dimension 1:
+    /// let parts = value.split(numSplits: 3, alongAxis: 1)
+    /// parts[0] // has shape [5, 10]
+    /// parts[1] // has shape [5, 10]
+    /// parts[2] // has shape [5, 10]
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - numSplits: Number of splits to create.
+    ///   - axis: Dimension along which to split this tensor. Negative values wrap around.
+    ///
+    /// - Precondition: `numSplits` must divide the size of dimension `axis` evenly.
+    /// - Precondition: `axis` must be in the range `[-rank, rank)`, where `rank` is the rank of the
+    ///   provided tensors.
+    ///
+    /// - Returns: Array containing the tensors parts.
+    @inlinable
+    @differentiable(vjp: _vjpSplit(numSplits:alongAxis:) where Scalar: TensorFlowFloatingPoint)
+    func split(numSplits: Int, alongAxis axis: Int = 0) -> [Tensor] {
```

Comment:
Swift APIs generally do not use a "num" prefix.
```suggestion
    func split(count: Int, alongAxis axis: Int = 0) -> [Tensor] {
```